BigBaseV2/src/memory/byte_patch.hpp

```diff
@@ -2,31 +2,64 @@
 
 namespace memory
 {
-	struct byte_patch
+	class byte_patch
 	{
-		void* address;
-		std::vector<uint8_t> original_bytes;
-
+	public:
 		template <typename TAddr>
-		static byte_patch make(TAddr address, std::remove_pointer_t<std::remove_reference_t<TAddr>> value)
+		byte_patch(TAddr address, std::remove_pointer_t<std::remove_reference_t<TAddr>> value)
+			: m_address(address)
 		{
-			byte_patch p;
-
-			p.address = address;
+			constexpr auto size = sizeof(std::remove_pointer_t<std::remove_reference_t<TAddr>>);
+			m_original_bytes.resize(size);
+			memcpy(m_original_bytes.data(), m_address, size);
 
-			constexpr size_t size = sizeof(std::remove_pointer_t<std::remove_reference_t<TAddr>>);
+			*address = value;
+		}
 
-			p.original_bytes.resize(size);
-			memcpy(p.original_bytes.data(), p.address, size);
+		virtual ~byte_patch()
+		{
+			memcpy(m_address, m_original_bytes.data(), m_original_bytes.size());
+		}
 
-			*address = value;
+		/// <summary>
+		/// Does not guarantee restoration of bytes, only after the object is deconstructed will the bytes be restored.
+		/// </summary>
+		void restore() const
+		{
+			if (const auto it = std::find(m_patches.begin(), m_patches.end(), this); it != m_patches.end())
+			{
+				m_patches.erase(it);
+			}
+		}
```

Comment:
I can probably call `it.reset()` on the iterator to guarantee the patch is restored.

Reply:
Done ✔️